AK/StdLibExtraDetails.h

```diff
@@ -142,6 +142,13 @@ inline constexpr bool IsFunction<Ret(Args...) const volatile&&> = true;
 template<class Ret, class... Args>
 inline constexpr bool IsFunction<Ret(Args..., ...) const volatile&&> = true;
 
+template<class>
+inline constexpr bool IsBlockClosure = false;
+template<class Ret, class... Args>
+inline constexpr bool IsBlockClosure<Ret (^)(Args...)> = true;
+template<class Ret, class... Args>
+inline constexpr bool IsBlockClosure<Ret (^)(Args..., ...)> = true;
```

Comment:
🤔 is this always available? probably needs some ifdef?

Reply:
Nope. You need -fblocks or to be compiling objective c. When targeting macOS blocks are enabled by the clang driver though, in the llvm toolchain files. And gcc doesn't support them at all. :/

Reply:
Yeah I guess I can go ifdeffing. I'll have to check if fblocks in clang gives us a processor macro or a `__has_feature` to check.

Reply:
It's `__has_feature(blocks)`